paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -96,50 +100,66 @@ def main() -> None:
     secret_provider_name = system_paasta_config.get_secret_provider_name()
     vault_cluster_config = system_paasta_config.get_vault_cluster_config()
     kube_client = KubeClient()
+    services_to_k8s_namespaces: Mapping[str, set] = defaultdict(set)
+    for service in args.service_list:
+        for instance_type in INSTANCE_TYPES:
+            instances = get_service_instance_list(
+                service=service,
+                instance_type=instance_type,
+                cluster=cluster,
+                soa_dir=args.soa_dir,
+            )
+            if instances:
+                services_to_k8s_namespaces[service].add(
+                    get_instance_type_to_k8s_namespace()[instance_type]
+                )
```

Comment:
should this be a function? one benefit we'd get from this being a function is that we could hide the defaultdict in the function and return a normal dict (so that we don't leave a trap for someone in the future when they thought they were working on a dict but were actually working on a defaultdict)

Reply:
I've refactored it into a function that returns a normal dict, not sure if I like either approach better